sechub-scan/src/test/java/com/mercedesbenz/sechub/domain/scan/ScanTypeCountTest.java

```diff
@@ -120,4 +155,40 @@ void compareTo_must_return_negative_value_bacause_scanTypeCountA_bigger_scanType
         /* test */
         assertTrue(0 > compareResult);
     }
+
+    @Test
+    void when_highSeverityCount_equals_MAX_VALUE_then_after_increasing_it_must_be_less_than_0() {
+        /* prepare */
+        scanTypeCount.highSeverityCount = Integer.MAX_VALUE;
```

Comment:
You would have to test for Long.MAX_VALUE and count it up by one. My expectation would be to get an Exception to avoid overflows.

The point of this test is to see how the software behaves at edge cases. Surly, `long` is a large value. However, the question of this test is, what would happen if it ever reaches the maximum? Does it wrap-around and is it prone to [CWE-128](https://cwe.mitre.org/data/definitions/128.html) or does it throw an exception?

Reply:
@Jeeppler Should I change the fields ___SeverityCount type from int to long for this test?

Reply:
No, long is fine. However, you should then test with Long.MAX_VALUE:

~~~
    @Test
    void when_highSeverityCount_equals_MAX_VALUE_then_after_increasing_it_must_be_less_than_0() {
        /* prepare */
        scanTypeCount.highSeverityCount = Long.MAX_VALUE;
      
       /* execute */
      scanTypeCount.incrementHighSeverityCount();

      /* test */
      assertTrue(0 > scanTypeCount.getHighSeverityCount());
}
~~~

The assertion will probably be false.

The nice thing about using methods is, that you can check in the `incrementHighSeverityCount()` method before counting up, that the value is less or equal to  `Long.MAX_VALUE` and if is more than that, you throw a Runtime exception (e.g. IllegalStateException or your own).

Reply:
@Jeeppler 
I changed my code according to your recommendation, but the assertion still returns true.
Please have a look at my last commit. I probably misunderstood you.

Reply:
Done